src/dialects/mysql/index.ts

```diff
@@ -54,6 +54,7 @@ export class MysqlDialect extends AbstractDialect {
       jsonOperations: true,
       REGEXP: true,
       globalTimeZoneConfig: true,
+      maxExecutionTimeHint: true,
```

Comment:
does mariadb support this too?

Reply:
MariaDB support functionally same thing([max_statement_time](https://mariadb.com/kb/en/aborting-statements/)), but it's a little bit diferrent.
1. unit time: `second` vs `millisecond`
2. name: `max_statement_time` vs `max_execution_time`
3. syntax: `SET STATEMENT max_statement_time=100 FOR SELECT ~` vs `SELECT /*+ MAX_EXECUTION_TIME(N) */ FROM ~`

Reply:
Ok then this PR can be exclusively for mysql

Reply:
Maybe we should implement this for `mariadb` as well in this PR? Even though it's `max_statement_time` I think that the same option would be fine to use. Possibly making this an object since mysql only supports this for `SELECT` queries but mariadb for different queries. (and then we'll just do the `SELECT` support in this PR to keep the scope limited)

Reply:
I'm OK with the actual implementation being done in a separate PR, but in preparation for that, I think `selectFromTableFragment` should call a new protected method `this._getSelectOptimizerHints` (returns `''` by default) and move the actual logic inside of `MySqlQueryGenerator`. MariaDB will need to replace it to return `''` since they use a different syntax.

MariaDB supports it on all queries, but right now we're only considering it for `SELECT`. Maybe name it `selectMaxExecutionTimeHint`? We can always change the flag if we decide to support other statements. 

Or alternatively:

```ts
// supported on all queries
maxExecutionTimeHint: true

// unsupported
maxExecutionTimeHint: true

// supported on these specific queries
maxExecutionTimeHint: ['select'],
// or?
maxExecutionTimeHint: { select: true },
```

Reply:
Yeah, I was thinking about `maxExecutionTimeHint: { select: true },`

Reply:
Then should we keep this PR for mysql and implement for mariadb in a separate PR?

Reply:
Yeah, some preparation for mariadb is done in this PR but the actual implementation in a separate PR.
So the `_getSelectOptimizerHints` that @ephys mentioned is good for this PR, just as updating `maxExecutionTimeHint` to accept an object `{ select: true }`

Reply:
I implemented `_getSelectOptimzerHints` returns 'SELECT` by default. Because in mysql, optimizer hint locate after of 'SELECT' but in mariadb locate before of 'SELECT'.